### ISSUE TITLE
Support for template language independent includes

### ISSUE THREAD

**Issue by jakearchibald** [11ty/eleventy#625]
Right now I can include content via features in particular template languages:

```njk
{% include "whatever.njk" %}
```

But, it'd be great if this could be first-class in 11ty, so I could do something similar in any template language that supports shortcodes.

This would allow me to include within JS templates:

```js
module.exports = function() {
  return `
    <p>yo</p>
    ${this.include('whatever.11ty.js')}
  `;
}
```

And even include different types of template:

```njk
{% inc "whatever.md" %}
```

(inc rather than include to avoid the clash with nunjuck's existing tag)

```js
module.exports = function() {
  return `
    <p>yo</p>
    ${this.include('whatever.njk')}
  `;
}
```

**Comment by jakearchibald** [comment]
This is a bit of a pain to hand-roll, as you end up duplicating the logic to handle the various export patterns, and need to make sure `this` is passed through. It'd also be nice to support data.

**Comment by zachleat** [comment]
Adding this one to the enhancement queue, sorry about the delay here!

> This repository is now using [lodash style issue management](https://twitter.com/samselikoff/status/991395669016436736) for enhancements. This means enhancement issues will now be closed instead of leaving them open.

> View [the enhancement backlog here](https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+). Don’t forget to upvote the top comment with 👍!

**Comment by zachleat** [comment]
Sorry for the super late response but the renderFile shortcode as part of the Render Plugin solved this (first shipped with v1.0.0) https://www.11ty.dev/docs/plugins/render/#render-file-shortcode
